CenterForOpenScience/osf.io
# Feature/Allow OAuth application client secret reset by PATCH [ENG-134] #8919

## Purpose
A cleaner way to reset the client secret for an OAuth application via the API would be to PATCH the developer-app resource with the client_secret property set to null. This would be more intuitive that POSTing to a special endpoint and would make the front end simpler.

Note: adds new API version

## Changes
- Deprecate old ApplicationReset view
- allow patching client_secret as null in ApiOAuth2ApplicationDetailSerializer
- Tests for deprecated view and new route



## QA Notes
Resetting your client secret uses v2, so verify that this still works.
![Screen Shot 2019-04-02 at 2 03 12 PM](https://user-images.githubusercontent.com/9755598/55429119-4e3ee480-5550-11e9-80af-fa1306163814.png)

Even though this PR adds a new endpoint, the existing page is calling the deprecated endpoint.  This should all still work as-is.  Front end will later update to call new endpoint.

## Documentation
It doesn't look like any of the current application API endpoints are currently in the documentation, so no updates needed until we decide to document the applications endpoints.



## Side Effects
None anticipated


## Ticket

https://openscience.atlassian.net/browse/ENG-134
